renewCerts.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# A script to renew certificates with certbot and send chat message with result
+
+# Example usage, expected within crontab:
+# 0 18 * * SUN /home/deploy/renewCerts.sh
+
+# Requirements:
+# certbot (installed on the host, not in a container),
+# nginx already configured to point to letsencrypt certificates, and
+# Environment variables in /home/deploy/.env:
+# - REVERSE_PROXY_CONTAINER_USER
+# - WEB_SERVER_HOSTNAME
+# - AUTH_SERVER_HOSTNAME
+# - ZULIP_BASE_URL
+# - ZULIP_BOT_EMAIL_ADDRESS
+# - ZULIP_BOT_API_KEY
+# - ZULIP_STREAM
+# - ZULIP_TOPIC
+
+set -eo pipefail
+
+test -x "$(which certbot)"
+. /home/deploy/.env
+test ! -z "$WEB_SERVER_HOSTNAME"
+test ! -z "$REVERSE_PROXY_CONTAINER_USER"
+test ! -z "$AUTH_SERVER_HOSTNAME"
+test ! -z "$ZULIP_BASE_URL"
+test ! -z "$ZULIP_BOT_EMAIL_ADDRESS"
+test ! -z "$ZULIP_BOT_API_KEY"
+test ! -z "$ZULIP_STREAM"
+test ! -z "$ZULIP_TOPIC"
+
+set +eo pipefail
+
+function fin() {
+    # Exit. Also notify of the result via chat if an API key is present.
+    exit_code=0
+    error_message=$1
+    message="✅ Certificate renewals (or checks) for https://${AUTH_SERVER_HOSTNAME} and https://${WEB_SERVER_HOSTNAME} succeeded."
+
+    if test ! -z "$error_message"; then
+        exit_code=1
+        message="❌ Certificate renewals (or checks) for https://${AUTH_SERVER_HOSTNAME} or https://${WEB_SERVER_HOSTNAME} FAILED: ${error_message}"
+    fi
+
+    if test ! -z "$ZULIP_BOT_API_KEY"; then
+	curl -X POST ${ZULIP_BASE_URL}/api/v1/messages \
+	     -u ${ZULIP_BOT_EMAIL_ADDRESS}:${ZULIP_BOT_API_KEY} \
+	     --data-urlencode type=stream \
+	     --data-urlencode to=${ZULIP_STREAM} \
+	     --data-urlencode topic=${ZULIP_TOPIC} \
+	     --data-urlencode "content=${message}"
+    fi
+
+    exit $exit_code
+}
+
+certbot certonly -n --domain=$AUTH_SERVER_HOSTNAME --standalone --keep-until-expiring \
+    || fin $(tail -n 10 /var/log/letsencrypt/letsencrypt.log)
+certbot certonly -n --domain=$WEB_SERVER_HOSTNAME --standalone --keep-until-expiring \
+    || fin $(tail -n 10 /var/log/letsencrypt/letsencrypt.log)
+# Make sure the reverse proxy user has read/execute privileges to the files
+chown -R root:$REVERSE_PROXY_CONTAINER_USER /etc/letsencrypt/{live,archive} \
```

Comment:
If we're leaving the ownership alone and just changing the group, let's rename the variable to `REVERSE_PROXY_CONTAINER_GROUP` and call `chgrp(1)` instead:

```suggestion
chgrp -R $REVERSE_PROXY_CONTAINER_GROUP /etc/letsencrypt/{live,archive} \
```

Reply:
+1 to the `chgrp` but we use the `REVERSE_PROXY_CONTAINER_USER` as the user that runs `nginx`. The idea is to give read access to that `nginx` user and nobody else.

Reply:
Reading between the lines, though, there is no guarantee that the `gid` matches the `uid`, so we should separate `GROUP` from `USER`.